Wrong "unused nowarn" warnings when there are errors

```scala
import annotation.nowarn
object T {
  @deprecated def f = 1
  def t1 = /
  @nowarn def t2 = f
}
```

Compiling this with `-Wunused:nowarn` gives an incorrect warning:

```
Test.scala:4: error: not found: value /
  def t1 = /
           ^
Test.scala:5: warning: @nowarn annotation does not suppress any warnings
  @nowarn def t2 = f
   ^
1 warning
1 error
```

The reason is that the error is reported during typer, which makes the compiler skip later phases. So refchecks doesn't run and the `@nowarn` annotation doesn't silence the deprecation warning.

--- a/src/compiler/scala/tools/nsc/Global.scala
+++ b/src/compiler/scala/tools/nsc/Global.scala
@@ -1562,12 +1562,11 @@ class Global(var currentSettings: Settings, reporter0: Reporter)
         if (settings.YstatisticsEnabled && settings.Ystatistics.contains(phase.name))
           printStatisticsFor(phase)
 
-        if (!globalPhase.hasNext || reporter.hasErrors)
-          runReporting.warnUnusedSuppressions()
-
         advancePhase()
       }
       profiler.finished()
+
+      runReporting.runFinished(hasErrors = reporter.hasErrors)
 
       reporting.summarizeErrors()
 

--- a/src/compiler/scala/tools/nsc/Reporting.scala
+++ b/src/compiler/scala/tools/nsc/Reporting.scala
@@ -80,11 +80,11 @@ trait Reporting extends internal.Reporting { self: ast.Positions with Compilatio
     def suppressionExists(pos: Position): Boolean =
       suppressions.getOrElse(pos.source, Nil).exists(_.annotPos.point == pos.point)
 
-    def warnUnusedSuppressions(): Unit = {
-      // if we stop before typer completes (errors in parser, Ystop), report all suspended messages
+    def runFinished(hasErrors: Boolean): Unit = {
+      // report suspended messages (in case the run finished before typer)
       suspendedMessages.valuesIterator.foreach(_.foreach(issueWarning))
-      // scaladoc doesn't run all phases, so not all warnings are emitted
-      if (settings.warnUnusedNowarn && !settings.isScaladoc)
+      // report unused nowarns only if all all phases are done. scaladoc doesn't run all phases.
+      if (!hasErrors && settings.warnUnusedNowarn && !settings.isScaladoc)
         for {
           source <- suppressions.keysIterator.toList
           sups   <- suppressions.remove(source)

--- a/test/files/neg/t12433.check
+++ b/test/files/neg/t12433.check
@@ -1,0 +1,4 @@
+t12433.scala:5: error: not found: value /
+  def t1 = /
+           ^
+1 error

--- a/test/files/neg/t12433.scala
+++ b/test/files/neg/t12433.scala
@@ -1,0 +1,7 @@
+// scalac: -Wunused:nowarn
+import annotation.nowarn
+object T {
+  @deprecated def f = 1
+  def t1 = /
+  @nowarn def t2 = f
+}

